### PR TITLE
Change ARC secureshield example to non-multithreading mode

### PR DESCRIPTION
**Pull request by YuguoWH**
Modify ARC secureshield support: 
secureshield will use non-multithreading mode, also modify the sample to fit this.

### CODE REVIEW

File: samples/boards/arc_secure_services/em_starterkit_em7d_normal_defconfig [hunk 1]
```diff
@@ -16,4 +18,5 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_GPIO=y
 CONFIG_ARC_MPU_ENABLE=y
 CONFIG_ARC_HAS_SECURE=y
-CONFIG_TRUSTED_EXECUTION_NONSECURE=y
+CONFIG_TRUSTED_EXECUTION_SECURE=y
```
Comment (ruuddw) [inline samples/boards/arc_secure_services/em_starterkit_em7d_normal_defconfig:21]: Why is the 'normal mode' part of the application changed to 'secure'?

File: samples/boards/arc_secure_services/nsim_sem_normal_defconfig [hunk 2]
```diff
@@ -7,6 +9,7 @@ CONFIG_BUILD_OUTPUT_BIN=n
 CONFIG_PRINTK=y
 CONFIG_ARCV2_INTERRUPT_UNIT=y
 CONFIG_ARCV2_TIMER=y
+CONFIG_ARC_MPU_ENABLE=y
```
Comment (ruuddw) [inline samples/boards/arc_secure_services/nsim_sem_normal_defconfig:12]: Is the MPU now supported in the 'normal' image?

Comment (YuguoWH) [reply samples/boards/arc_secure_services/nsim_sem_normal_defconfig:12]: Yes, but maybe I should undo this modification until future pull request to actually support MPU?

**Comment by ruuddw** [comment]
Great to have no-multithreading support. I see only a 1 line functional change to support this, wouldn't there be some more opportunities for simplifying the ARC architecture part when multi-threading is disabled (e.g. simpler stack management/guard checking or so). Though, these could be left as future optimizations for now.

Did you check if the minimal guaranteed features for no-multithreading (inits, interrupts, UART/GPIO/Timer, see documentation) are indeed working correctly without any changes?

File: samples/boards/arc_secure_services/nsim_sem_normal_defconfig (hunk 2 above)
Comment (ruuddw) [reply samples/boards/arc_secure_services/nsim_sem_normal_defconfig:12]: Yes, I think it is better to focus this PR on support for no-multithreading only.

**Pull request closed without merging**